vividus-plugin-kafka/src/main/java/org/vividus/bdd/steps/kafka/KafkaSteps.java

```diff
@@ -184,27 +209,44 @@ public void stopKafkaListener()
      * @param variableName   the variable name to store the messages. The messages are accessible via zero-based index,
      *                       e.g. `${my-var[0]}` will return the first received message.
      */
-    @When("I $queueOperation consumed Kafka messages to $scopes variable `$variableName`")
-    public void processKafkaMessages(QueueOperation queueOperation, Set<VariableScope> scopes,
+    @When("I $queueOperation consumed `$consumerKey` Kafka messages to $scopes variable `$variableName`")
+    public void processKafkaMessages(QueueOperation queueOperation, String consumerKey, Set<VariableScope> scopes,
             String variableName)
     {
-        bddVariableContext.putVariable(scopes, variableName, queueOperation.performOn(testContext.get(MESSAGES_KEY)));
+        bddVariableContext.putVariable(scopes, variableName, queueOperation.performOn(
+                testContext.get(messagesKey(consumerKey))));
+    }
+
+    private String messagesKey(String consumerKey)
+    {
+        return consumerKey + "kafkaMessages";
```

Comment:
I think the key should be more strict to avoid possible collisions (the best approach is to rely on some class name which is unique in scope of the plugin)